### PR TITLE
refactor config

### PR DESCRIPTION
**Pull request by grimoire**
Decoupling config and model.

### CODE REVIEW

File: lmdeploy/pytorch/configurations/builder.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from abc import ABC
+
+from lmdeploy.utils import get_logger
+
+logger = get_logger('lmdeploy')
+
+
+class AutoModelConfigBuilder(ABC):
+
+    _sub_classes = list()
```
Comment (RunningLeon) [inline lmdeploy/pytorch/configurations/builder.py:11]: why not use dict in here for quickly querying?

Comment (grimoire) [reply lmdeploy/pytorch/configurations/builder.py:11]: What key?

Comment (RunningLeon) [reply lmdeploy/pytorch/configurations/builder.py:11]: string of class name?

Comment (grimoire) [reply lmdeploy/pytorch/configurations/builder.py:11]: It is meaningless, we would only use the class when `condition` is true.

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**